custom_components/lennoxs30/__init__.py

```diff
@@ -75,15 +82,48 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+async def async_setup(
+    hass : HomeAssistant,
+    config : ConfigType
+):
+    """Import config as config entry."""
+    hass.data[DOMAIN] = {}
+    _LOGGER.warning(
+        "Configuration of the LennoxS30 platform in YAML is deprecated "
+        "and will be removed; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_NAME: "S30",
+                CONF_EMAIL: config.get(DOMAIN).get(CONF_EMAIL),
+                CONF_PASSWORD:  config.get(DOMAIN).get(CONF_PASSWORD),
+                CONF_HOSTS:  config.get(DOMAIN).get(CONF_HOSTS),
+                CONF_SCAN_INTERVAL:  config.get(DOMAIN).get(CONF_SCAN_INTERVAL),
+                CONF_FAST_POLL_INTERVAL:  config.get(DOMAIN).get(CONF_FAST_POLL_INTERVAL),
+                CONF_ALLERGEN_DEFENDER_SWITCH:  config.get(DOMAIN).get(CONF_ALLERGEN_DEFENDER_SWITCH),
+                CONF_APP_ID:  config.get(DOMAIN).get(CONF_APP_ID),
+                CONF_INIT_WAIT_TIME:  config.get(DOMAIN).get(CONF_INIT_WAIT_TIME),
+                CONF_CREATE_SENSORS:  config.get(DOMAIN).get(CONF_CREATE_SENSORS),
+            },
```

Comment:
Need to add the power inverter into this migraiton:
                CONF_CREATE_INVERTER_POWER: config.get(DOMAIN).get(
                    CONF_CREATE_INVERTER_POWER